### PR TITLE
feat(web): ImageBitmap

### PR DESCRIPTION
**Pull request by crowlKats**
implements the `ImageBitmap` class and `createImageBitmap` function from https://html.spec.whatwg.org/multipage/imagebitmap-and-animations.html.

Prerequisite for #5701 

This API by itself is useless without OffscreenCanvas and/or integration into WebPGU, since accessing the data of an `ImageBitmap` directly is not possible.

- [ ] ~~enable some WPTs~~ no WPTs can be enabled due to all relevant WPTs requiring OffscreenCanvas
- [x] typings
- [x] lazy load

### CODE REVIEW

File: ext/canvas/01_image.js [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+    );
+    imageBitmap[_bitmapData] = processedImage.data;
+    imageBitmap[_width] = processedImage.outputWidth;
+    imageBitmap[_height] = processedImage.outputHeight;
+    return PromiseResolve(imageBitmap);
+  }
+  if (ObjectPrototypeIsPrototypeOf(BlobPrototype, image)) {
+    return (async () => {
+      const data = await image.arrayBuffer();
+      // TODO: 2. mimesniff (we only support png)
```
Comment (crowlKats) [inline ext/canvas/01_image.js:418]: TODO

**Review by littledivy — COMMENTED** [review]
Woah, nice! 

ImageBitmap can integrate with WebGPU's `copyExternalImageToTexture` - very useful for texture rendering.

**Comment by crowlKats** [comment]
@littledivy yes, however this new "canvas" extension will depend on the webgpu extension, so we will need to be careful when adding support for such, and will be tricky especially since webgpu implementation also lives in wgpu.
definitively a usecase to consider, however I'd say lets focus first on the direct usecase of OffscreenCanvas and investigate this later on.

File: runtime/js/98_global_scope_shared.js [hunk 2]
```diff
@@ -31,11 +31,67 @@ import * as messagePort from "ext:deno_web/13_message_port.js";
...
+    enumerable: true,
+    configurable: true,
+  };
+}
+function loadImage() {
+  if (!image) {
+    image = op_lazy_load_esm("ext:deno_canvas/01_image.js");
+  }
+}
+
```
Comment (crowlKats) [inline runtime/js/98_global_scope_shared.js:94]: quite a bit duplicate with webgpu. we should look into having a better solution that requires less setup for each lazy loaded file

**Review by littledivy — COMMENTED** [review]
Are there WPT we can enable or unit tests that can be added?

**Comment by crowlKats** [comment]
@littledivy from the PR description:
> no WPTs can be enabled due to all relevant WPTs requiring OffscreenCanvas
I'll add a very rudimentary unit test though

File: ext/canvas/lib.rs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+#[op2]
+#[serde]
+fn op_image_process(
+  #[buffer] buf: &[u8],
+  #[serde] args: ImageProcessArgs,
+) -> Result<ToJsBuffer, AnyError> {
+  let view =
+    RgbaImage::from_vec(args.width, args.height, buf.to_vec()).unwrap();
```
Comment (littledivy) [inline ext/canvas/lib.rs:48]: The full allocation here will hurt performance. I think you can create a view into the JS slice:
```js
 let view: ImageBuffer<Rgba<u8>, &[u8]> = ImageBuffer::from_raw(args.witdh, args.height, buf).unwrap();
```

File: ext/canvas/lib.rs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  let mut surface = RgbaImage::new(args.surface_width, args.surface_height);
+
+  image::imageops::overlay(&mut surface, &view, args.input_x, args.input_y);
```
Comment (littledivy) [inline ext/canvas/lib.rs:51]: We should add a fast path which does not allocate this extra surface when surface dimenstions == view dimenstion (which I assume is always the case unless source rectangle is specified from the API).

Can you add a TODO for this?

File: ext/canvas/lib.rs (hunk 3 above)
Comment (crowlKats) [reply ext/canvas/lib.rs:48]: sadly I attempted this already, but it wont accept a slice no matter what (even your code, oddly enough, errors out with it wanting a vec), which is weird given that the docs for `from_raw` state that a slice can be passed.

File: ext/canvas/lib.rs (hunk 4 above)
Comment (crowlKats) [reply ext/canvas/lib.rs]: I implemented it, PTAL

**Review by littledivy — APPROVED** [review]
LGTM, nice work

**Review by lucacasonato — APPROVED** [review]
LGTM

**Pull request merged**